corehq/apps/users/models.py

```diff
@@ -1139,6 +1139,33 @@ def get(stale, raise_if_none):
         else:
             return None
 
+    @classmethod
+    @skippable_quickcache(['username'], skip_arg='strict')
+    def get_deleted_by_username(cls, username, strict=True):
+        def get(stale, raise_if_none):
+            result = cls.get_db().view('deleted_users/by_username',
+                                       key=username,
+                                       include_docs=True,
+                                       reduce=False,
+                                       stale=stale if not strict else None,
+                                       )
+            return result.one(except_all=raise_if_none)
```

Comment:
it will load all into memory and check to make sure it's 1, though for this i actually don't think it matters since we only expect one to return from the view